lib/ansible/modules/cloud/amazon/ec2.py

```diff
@@ -1001,7 +1022,7 @@ def create_instances(module, ec2, vpc, override_count=None):
 
     # group_id and group_name are exclusive of each other
     if group_id and group_name:
-        module.fail_json(msg = str("Use only one type of parameter (group_name) or (group_id)"))
```

Comment:
I think this can safely be made into a mutually_exclusive parameter on AnsibleModule, since this `if` seems to be repeated a couple times.